doppler/resource_webhook.go

```diff
@@ -0,0 +1,247 @@
+	client := m.(APIClient)
+
+	var diags diag.Diagnostics
+	project := d.Get("project").(string)
+	url := d.Get("url").(string)
+	enabled := d.Get("enabled").(bool)
+	secret := d.Get("secret").(string)
+	payload := d.Get("payload").(string)
+
+	rawEnabledConfigs := d.Get("enabled_configs").(*schema.Set).List()
+	enabledConfigs := make([]string, len(rawEnabledConfigs))
+	for i, v := range rawEnabledConfigs {
+		enabledConfigs[i] = v.(string)
+	}
+
+	options := WebhookOptionalParameters{Secret: secret, WebhookPayload: payload, EnabledConfigs: enabledConfigs}
+
+	authConfigList := d.Get("authentication").([]interface{})
+
+	if len(authConfigList) > 0 {
+		authMap, ok := authConfigList[0].(map[string]interface{}) // schema allows only 1 item
+		if !ok {
+			return diag.FromErr(fmt.Errorf("unexpected type for authentication element: %T", authConfigList))
+		}
+
+		options.Auth = &WebhookAuth{
+			Type:     authMap["type"].(string),
+			Token:    authMap["token"].(string),
+			Username: authMap["username"].(string),
+			Password: authMap["password"].(string),
+		}
+	}
+
+	webhook, err := client.CreateWebhook(ctx, project, url, enabled, &options)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(webhook.Slug)
+
+	return diags
+}
+
+func resourceWebhookUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(APIClient)
+
+	var diags diag.Diagnostics
+	slug := d.Id()
+	project := d.Get("project").(string)
+
+	if d.HasChange("enabled") {
+		if d.Get("enabled").(bool) {
+			_, err := client.EnableWebhook(ctx, project, slug)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			_, err := client.DisableWebhook(ctx, project, slug)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	url := d.Get("url").(string)
+	enabledConfigs := []string{}
+	disabledConfigs := []string{}
+	if d.HasChange("enabled_configs") {
+		oldRawEnabledConfigs, newRawEnabledConfigs := d.GetChange("enabled_configs")
+		oldRawEnabledConfigsArr := oldRawEnabledConfigs.(*schema.Set).List()
+		newRawEnabledConfigsArr := newRawEnabledConfigs.(*schema.Set).List()
+		oldEnabledConfigsMap := make(map[string]string)
+		for _, v := range oldRawEnabledConfigsArr {
+			oldEnabledConfigsMap[v.(string)] = v.(string)
+		}
+		newEnabledConfigsMap := make(map[string]string)
+		for _, v := range newRawEnabledConfigsArr {
+			newEnabledConfigsMap[v.(string)] = v.(string)
+		}
+
+		for _, v := range newRawEnabledConfigsArr {
+			if _, ok := oldEnabledConfigsMap[v.(string)]; !ok {
+				enabledConfigs = append(enabledConfigs, v.(string))
+			}
+		}
+
+		for _, v := range oldRawEnabledConfigsArr {
+			if _, ok := newEnabledConfigsMap[v.(string)]; !ok {
+				disabledConfigs = append(disabledConfigs, v.(string))
+			}
+		}
+	}
+
+	webhook, err := client.UpdateWebhook(ctx, project, slug, url, enabledConfigs, disabledConfigs)
```

Comment:
Related to earlier comment: We should update `secret`, `authentication`, and `payload` here